src/libAtomVM/nifs.c

```diff
@@ -3297,35 +3297,96 @@ static term nif_erlang_raise(Context *ctx, int argc, term argv[])
     return term_invalid_term();
 }
 
+static inline term get_option_key(term option, term *maybe_value)
+{
+    *maybe_value = term_invalid_term();
+    if (LIKELY(term_is_atom(option))) {
+        return option;
+    }
+
+    bool is_tuple = term_is_tuple(option) && term_get_tuple_arity(option) == 2;
+    if (UNLIKELY(!is_tuple)) {
+        return term_invalid_term();
+    }
+    term first = term_get_tuple_element(option, 0);
+    term second = term_get_tuple_element(option, 1);
+    if (UNLIKELY(!term_is_atom(first))) {
+        return term_invalid_term();
+    }
+
+    *maybe_value = second;
+    return first;
+}
+
 static term nif_ets_new(Context *ctx, int argc, term argv[])
 {
     UNUSED(argc);
-
     term name = argv[0];
-    VALIDATE_VALUE(name, term_is_atom);
-
     term options = argv[1];
+    VALIDATE_VALUE(name, term_is_atom);
     VALIDATE_VALUE(options, term_is_list);
 
-    term is_named = interop_kv_get_value_default(options, ATOM_STR("\xB", "named_table"), FALSE_ATOM, ctx->global);
-    term keypos = interop_kv_get_value_default(options, ATOM_STR("\x6", "keypos"), term_from_int(1), ctx->global);
-
-    if (term_to_int(keypos) < 1) {
-        RAISE_ERROR(BADARG_ATOM);
+    bool is_named = false;
+    bool private = false;
+    bool public = false;
+    bool is_duplicate_bag = false;
+    avm_int_t keypos = 1;
+
+    while (term_is_nonempty_list(options)) {
+        term head = term_get_list_head(options);
+        term value;
+        term key_atom = get_option_key(head, &value);
+        VALIDATE_VALUE(key_atom, term_is_atom);
+
+        switch (key_atom) {
+            case NAMED_TABLE_ATOM: {
```

Comment:
fun fact: on OTP it is not a proplist, so `ets:new(foo, [{named_table, true}]).` raises badarg, but `ets:new(fiz, [named_table])` doesn't.
So I think this can be handled with a simple change:
```c
switch (head) {
   case NAMED_TABLE_ATOM:
      ...
   default:
       VALIDATE_VALUE(head, term_is_tuple);
       term value;
       term key_atom = get_option_key(head, &value);
       VALIDATE_VALUE(key_atom, term_is_atom);
```

`ets:new(foo, [{named_table, true}]).` also, let's make sure in tests that fails as expected.